fdbclient/CommitTransaction.h

```diff
@@ -96,7 +96,18 @@ struct MutationRef {
 
 	template <class Ar>
 	void serialize( Ar& ar ) {
-		serializer(ar, type, param1, param2);
+		if (!ar.isDeserializing && type == ClearRange && equalsKeyAfter(param1, param2)) {
+			StringRef hold = param1;
+			param1 = StringRef();
+			serializer(ar, type, param2, param1);
+			param1 = hold;
+		} else {
+			serializer(ar, type, param1, param2);
+		}
+		if (ar.isDeserializing && type == ClearRange && param2 == StringRef() && param1[param1.size()-1] == '\x00') {
```

Comment:
Is it possible for param1 to have size 0 here? I guess the idea is that if `param2 == StringRef()` then param1 must be part of this special encoding or else it would be an inverted range.

In that case, can we assert that `param1.size() > 0 && param1[param1.size() - 1] == '\x00'` if `param2 == StringRef()`?

Reply:
good idea, I will add the assert